### PR TITLE
Add JS GeneratorFunction examples

### PR DESCRIPTION
**Pull request by NiedziolkaMichal**
Interactive examples for [GeneratorFunction](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/GeneratorFunction) and [AsyncGeneratorFunction](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGeneratorFunction).

image

image

### CODE REVIEW

File: live-examples/js-examples/functionasterisk/async-functionasterisk-function.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const AsyncGeneratorFunction = (async function* () {}).constructor;
```
Comment (Josh-Cena) [inline live-examples/js-examples/functionasterisk/async-functionasterisk-function.js]:
```suggestion
const AsyncGeneratorFunction = async function* () {}.constructor;
```

Prettier-style

**Review by Josh-Cena — APPROVED** [review]
LGTM; you may want to format it so the style is aligned with content (e.g. also use double quotes). These pages are very obscure so I won't worry much. We recently split the constructor pages from the main class pages—I'm almost sure it doesn't matter for interactive examples, but just as a heads-up

File: live-examples/js-examples/functionasterisk/async-functionasterisk-function.js (hunk 1 above)
Comment (NiedziolkaMichal) [reply live-examples/js-examples/functionasterisk/async-functionasterisk-function.js]: Thank you, that looks better

**Comment by bsmth** [comment]
Thanks, both, merging shortly 👍🏻

**Pull request merged**